dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeDestroyAPIImpl.java

```diff
@@ -0,0 +1,351 @@
+import com.dotcms.api.system.event.Payload;
+import com.dotcms.api.system.event.SystemEventType;
+import com.dotcms.api.system.event.Visibility;
+import com.dotcms.business.CloseDBIfOpened;
+import com.dotcms.business.WrapInTransaction;
+import com.dotcms.contenttype.model.event.ContentTypeDeletedEvent;
+import com.dotcms.contenttype.model.type.ContentType;
+import com.dotcms.exception.BaseRuntimeInternationalizationException;
+import com.dotcms.notifications.bean.NotificationLevel;
+import com.dotcms.notifications.bean.NotificationType;
+import com.dotcms.notifications.business.NotificationAPI;
+import com.dotcms.system.event.local.business.LocalSystemEventsAPI;
+import com.dotcms.util.ContentTypeUtil;
+import com.dotcms.util.I18NMessage;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.business.DotStateException;
+import com.dotmarketing.business.FactoryLocator;
+import com.dotmarketing.business.PermissionAPI;
+import com.dotmarketing.business.Role;
+import com.dotmarketing.common.db.DotConnect;
+import com.dotmarketing.db.DbConnectionFactory;
+import com.dotmarketing.db.HibernateUtil;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.business.ContentletDestroyDelegate;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
+import com.google.common.collect.Lists;
+import com.liferay.portal.model.User;
+import io.vavr.Lazy;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+public class ContentTypeDestroyAPIImpl implements ContentTypeDestroyAPI {
+
+    /**
+     * This method relocates all contentlets from a given structure to another structure And returns
+     * the new structure purged from fields
+     *
+     * @param source
+     * @param target
+     * @throws DotDataException
+     */
+    @WrapInTransaction
+    public void relocateContentletsForDeletion(final ContentType source, final ContentType target) throws DotDataException {
+
+        if (!source.getClass().equals(target.getClass())) {
+            throw new DotDataException(
+                    String.format(
+                            "Incompatible source and target ContentTypes. source class is (%s) target class is (%s)",
+                            source.getClass().getSimpleName(), target.getClass().getSimpleName()
+                    )
+            );
+        }
+
+        final DotConnect dotConnect = new DotConnect();
+
+        final String selectContentlets = "select c.inode from contentlet c where c.structure_inode =  ? \n";
+        dotConnect.setSQL(selectContentlets).addParam(source.inode());
+
+        String updateContentlet = String.format(
```

Comment:
it is just style but you can use 
final String updateContent = DbConnectionFactory.isMsSql()?xxxx:yyyy;

Reply:
Also you can use constants such MSSQL_XXXX, PG_YYYY